app/src/main/java/com/example/weathertracking/Activities/LocationSearchActivity.java

```diff
@@ -0,0 +1,488 @@
+
+    Button searchLocationsButton;
+    Button viewWeatherButton;
+    Switch favoriteSwitch;
+    SearchResultAdapter adapter;
+    RecyclerView recyclerView;
+    List<SearchCityObject> results;
+    TextView noResultsTextView;
+    TextView locationEditText;
+    ImageButton locationSearch;
+
+    String Provider;
+    LatLng currentLatLon;
+
+    //WeatherForecastCalls forecastObject;
+    FavoriteLocationObject favoriteLocationObject;
+    Boolean currentLocationIsUsed = false;
+    int selecteditemNum= -1;
+
+    LatLng selectedLocationCoord;
+    String selectedLocationName;
+
+    ArrayList<FavoriteLocationObject> locationsForMap;
+    CompoundButton.OnCheckedChangeListener switchListener;
+    BroadcastReceiver itemClickedReceiver;
+
+    BroadcastReceiver choosedLocationReceiver;
+    BroadcastReceiver locationReceiver;
+    BroadcastReceiver currentWeatherReceiver;
+    BroadcastReceiver searchCitiesReceiver;
+    BroadcastReceiver networkActionReceiver;
+
+    IntentFilter currentWeatherFilter;
+    IntentFilter searchCitiesFilter;
+    IntentFilter intentFilterLocation;
+    IntentFilter networkActionFilter;
+    Boolean isNetwork=true;
+
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_location_search);
+
+        locationSearch = findViewById(R.id.locationImage);
+        favoriteSwitch= findViewById(R.id.favoriteSwitch);
+        viewWeatherButton = findViewById(R.id.viewWeatherButton);
+        locationEditText = findViewById(R.id.locationTextView);
+        searchLocationsButton = findViewById(R.id.searchLocationsButton);
+        recyclerView = findViewById(R.id.resultsRecyclerView);
+        noResultsTextView = findViewById(R.id.noResultsTextView);
+
+        viewWeatherButton.setVisibility(View.INVISIBLE);
+        selecteditemNum=-1;
+
+
+
+
+        switchListener = new CompoundButton.OnCheckedChangeListener() {
+            @Override
+            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+                modifyFavorite(locationsForMap.get(selecteditemNum),getBaseContext());
+            }
+        };
+
+        favoriteSwitch.setOnCheckedChangeListener(switchListener);
+
+        locationEditText.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if(!currentLocationIsUsed) {
+                    viewWeatherButton.setVisibility(View.INVISIBLE);//kell e ide if vagy nem>
+
+                    favoriteSwitch.setChecked(false);
+                    favoriteSwitch.setClickable(false);
+                    if (locationEditText.getCurrentTextColor()==(Color.GREEN)) {
```

Comment:
The Color.Green is in brackets, why?

Reply:
Also, what happens if there will be a design change, and the color of the edittext wont be green ever, it will be purple instead?